CMakeLists.txt

```diff
@@ -110,6 +109,29 @@ endif()
 
 if (BUILD_TESTS)
   enable_testing()
+
+  # For some Linux distro versions there is an unused-result warning
+  # generated inside the Gtest code at a call to fwrite. Since Gtest sets -Werror
+  # internally the only way to make it compile without modifying its source is to disable the
+  # unused-result warnings for Gtest/Gmock only.
+
+  set( TMP_C_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
+  set( TMP_CXX_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
+  set( TMP_C_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+  set( TMP_CXX_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
```

Comment:
I don't like it here... maybe some push()/pop() kind of self-defined macro/function

Reply:
We already use CMAKE_C_FLAGS/CMAKE_CXX_FLAGS for non-trivial configuration, and these are not overridden by target_compile_options properties. To use the target_compile_options approach, the whole CMake build structure of our project would have to be refactored to use target_compile_options instead of CMAKE_C_FLAGS/CMAKE_CXX_FLAGS

Reply:
Done